src/base.ts

```diff
@@ -79,38 +75,73 @@ export const resolveServiceUrl = (service: Service): string => {
   throw new Error(`Missing endpoint configuration for "${service}" service`)
 }
 
-const authorizationHeader = () => {
-  if (!apiClient.tokens.accessToken) {
-    return null
+const getAuthorizationHeader = (token = null) => {
+  console.log("getAuthorizationHeader instance ", apiClient)
+  const accessToken = token || apiClient.accessToken.token
+  if (!accessToken) {
+    throw new Error(
+      "You tried to make an authenticated requests without providing an access token!"
+    )
   }
 
-  return `Bearer ${apiClient.tokens.accessToken}`
+  return { Authorization: `Bearer ${accessToken}` }
+}
+
+export interface RequestOptions {
+  isAuthorizedRequest?: boolean
+  accessToken?: string
+  method?: "GET" | "POST" | "PUT" | "PATCH" | "DELETE"
+  headers?: {}
 }
 
-export const fetchPath = async (
-  service: Service,
-  path: string,
-  options = {}
-) => {
-  const prefix = resolveServiceUrl(service)
-  const url = `${prefix}/${stripLeadingSlash(path)}`
+export const fetchPath = async ({
```

Comment:
Went with updating the interface over a polymorphic options strategy where the options could be a different arguement based on weather the current request has a body or not